### PR TITLE
Add an option to invite users to upgraded private rooms

### PR DESCRIPTION
**Pull request by turt2live**
The option doesn't show up on public room upgrades. We abuse the MultiInviter for this so we don't have to deal with failure.

This only adds the toggle switch to this dialog:
image

### CODE REVIEW

**Comment by aaronraimist** [comment]
What changed from 6 days ago? https://github.com/vector-im/riot-web/issues/11464#issuecomment-557580490

File: src/utils/MultiInviter.js [hunk 1]
```diff
@@ -154,6 +154,8 @@ export default class MultiInviter {
                     return;
                 }
 
+                console.error(err);
```
Comment (turt2live) [inline src/utils/MultiInviter.js]: rationale: developers saying "why is my invite failing"

**Comment by turt2live** [comment]
@aaronraimist nothing. This isn't my favourite fix at all because ideally we'd just transfer membership instead of hammering the server, but here we are.

File: src/SlashCommands.js [hunk 2]
```diff
@@ -154,70 +155,55 @@ export const CommandMap = {
...
+                        // We have to wait for the js-sdk to give us the room back so
+                        // we can more effectively abuse the MultiInviter behaviour
+                        // which heavily relies on the Room object being available.
+                        if (resp.invite) {
+                            checkForUpgradeFn = async (newRoom) => {
+                                // The upgradePromise should be done by the time we await it here.
+                                const {replacement_room: newRoomId} = await upgradePromise;
+                                if (newRoom.roomId !== newRoomId) return;
+
+                                const joinedMembers = room.getJoinedMembers()
```
Comment (bwindels) [inline src/SlashCommands.js]: nit: I guess ideally we'd also invite people that still have membership "invite".

**Comment by nadonomy** [comment]
Room upgrades

Updated design, making note to trigger the normal help modal, and implement the toggle interaction on the right for consistency with the rest of the app (ref create room). Copy for your clipboard:

```
Upgrade public room
Upgrading a room is an advanced action and is usually recommended when a room is unstable due to bugs, missing features or security vulnerabilities.

This usually only effects how the room is processed on the server. If you're having problems with your Riot, please report a bug.

You’ll upgrade this room from v4 to v5.
```

```
Upgrade private room
Upgrading a room is an advanced action and is usually recommended when a room is unstable due to bugs, missing features or security vulnerabilities.

This usually only effects how the room is processed on the server. If you're having problems with your Riot, please report a bug.

You’ll upgrade this room from v4 to v5.
```

**Comment by turt2live** [comment]
per conversation out of band, to make it a bit fairer to Bruno I'm going to merge this as-is and put the design through a subsequent PR right away.

**Pull request merged**